tests/org.eclipse.swt.tests/JUnit Tests/org/eclipse/swt/tests/junit/Test_org_eclipse_swt_browser_Browser.java

```diff
@@ -518,6 +519,86 @@ public void changing(LocationEvent event) {
 	for (int i = 0; i < 100; i++) browser.removeLocationListener(listener);
 }
 
+@Test
+public void test_isLocationForCustomText_setUrlAfterDisposedThrowsSwtException() {
+	Browser testBrowser = createBrowser(shell, SWT.NONE);
+	testBrowser.dispose();
+	assertThrows(SWTException.class, () -> testBrowser.isLocationForCustomText("about:blank"));
+}
+
+@Test
+public void test_isLocationForCustomText_isSetUrlNotCustomTextUrlAfterSetText() {
+	assumeFalse("behavior is not (yet) supported by Edge browser", isEdge);	// Edge doesn't fire a changed event if the URL is the same as the previous location.
+	URI url = getValidUrl();
+	AtomicBoolean locationChanged = new AtomicBoolean(false);
+	browser.addLocationListener(changedAdapter(event -> {
+			locationChanged.set(true);
+	}));
+
+	browser.setText("Custom text");
+	assertTrue("Time Out: The Browser didn't navigate to the URL", waitForPassCondition(locationChanged::get));
+	locationChanged.set(false);
+	browser.setUrl(url.toASCIIString());
+	shell.open();
+	assertTrue("Time Out: The Browser didn't navigate to the URL", waitForPassCondition(locationChanged::get));
+	assertEquals("Url is wrongly considered Custom Text Url", URI.create(browser.getUrl()), url);
+	assertFalse("The navigated URL is falsly indicated to be the custom text URL", browser.isLocationForCustomText(browser.getUrl()));
+}
+
+@Test
+public void test_isLocationForCustomText_isFirstSetTextURLStillCustomTextUrlAfterSetUrl() {
+	assumeFalse("behavior is not (yet) supported by Edge browser", isEdge);	// Edge doesn't fire a changed event if the URL is the same as the previous location.
+	AtomicBoolean locationChanged = new AtomicBoolean(false);
+	browser.addLocationListener(changedAdapter(event -> locationChanged.set(true)));
+
+	URI url = getValidUrl();
+	shell.open();
+	browser.setText("Custom text");
+	assertTrue(waitForPassCondition(locationChanged::get));
+	String firstUrl = browser.getUrl();
+	locationChanged.set(false);
+	browser.setUrl(url.toASCIIString());
+	assertTrue("Time Out: The Browser didn't navigate to the URL", waitForPassCondition(locationChanged::get));
+	assertTrue(browser.isLocationForCustomText(firstUrl));
+	assertFalse(browser.isLocationForCustomText(browser.getUrl()));
+}
+
+private URI getValidUrl() {
+	String pluginPath = System.getProperty("PLUGIN_PATH");
+	testLogAppend("PLUGIN_PATH: " + pluginPath);
```

Comment:
I still do not understand the idea of this logging. The `testLogAppend()` seems to be used for writing to a local log that is evaluated only at very specifiy places (e.g. `validateTitleChanged()`). Is the value ever used in the related test cases? Otherwise, e.g., if this is only for local debugging purposes by manually inspecting the written variable in a debugger, you may also just set a breakpoint here and take a look at the original variable (`pluginPath`).
Don't get me wrong: writing to this log does not really hurt, but it leaves future maintainers future with the question what the intent of executing this operation was (as it not obviously useful).

Reply:
I just used one of the fragments which was generating the URL in one of the tests. To me, it has no use. So I can also remove it.